src/services/user.service.js

```
@@ -0,0 +1,9 @@
+import { User } from '../models/user.js';
+
+const createUser = async (name) => {
+  return User.create({ name });
+};
```

To improve separation of concerns, the logic for checking if a user already exists should be moved from the controller into this function. The `createUser` function should first query the database to see if a user with the given `name` already exists. If so, it should throw an error, which the controller can then catch and translate into a 409 Conflict response. This makes the service layer responsible for all business logic related to user creation.